dd-library-php-setup.php

```diff
@@ -775,136 +780,311 @@ function build_known_command_names_matrix()
-; setcap utility.
-;datadog.trace.retain_thread_capabilities = Off
+    if ($formattedMissingProperties !== '') {
+        if (false === file_put_contents($iniFilePath, $iniFileContent . $formattedMissingProperties)) {
+            print_error_and_exit("Cannot add additional settings to the INI file $iniFilePath");
+        }
+    }
+}
 
-EOD;
+/**
+ * Returns array of associative arrays with the following keys:
+ *   - name (string): the setting name;
+ *   - default (string): the default value;
+ *   - commented (bool): whether this setting should be commented or not when added;
+ *   - description (string|string[]): A string (or an array of strings, each representing a line) that describes
+ *                                    the setting.
+ *
+ * @param string $requestInitHookPath
+ * @return array
+ */
+function get_ini_settings($requestInitHookPath)
+{
+    // phpcs:disable Generic.Files.LineLength.TooLong
+    return [
+        [
+            'name' => 'extension',
+            'default' => 'ddtrace.so',
+            'commented' => false,
+            'description' => 'Enables or disables tracing (set by the installer, do not change it)',
+        ],
+        [
+            'name' => 'datadog.trace.request_init_hook',
+            'default' => $requestInitHookPath,
+            'commented' => false,
+            'description' => 'Path to the request init hook (set by the installer, do not change it)',
+        ],
+        [
+            'name' => 'datadog.trace.enabled',
+            'default' => 'On',
+            'commented' => true,
+            'description' => 'Enables or disables tracing. On by default',
+        ],
+        [
+            'name' => 'datadog.trace.cli_enabled',
+            'default' => 'Off',
+            'commented' => true,
+            'description' => 'Enable or disable tracing of CLI scripts. Off by default',
+        ],
+        [
+            'name' => 'datadog.trace.auto_flush_enabled',
+            'default' => 'Off',
+            'commented' => true,
+            'description' => 'For long running processes, this setting has to be set to On',
+        ],
+        [
+            'name' => 'datadog.trace.generate_root_span',
+            'default' => 'On',
+            'commented' => true,
+            'description' => 'For long running processes, this setting has to be set to Off',
+        ],
+        [
+            'name' => 'datadog.trace.debug',
+            'default' => 'Off',
+            'commented' => true,
+            'description' => 'Enables or disables debug mode.  When On logs are printed to the error_log',
+        ],
+        [
+            'name' => 'datadog.trace.startup_logs',
+            'default' => 'On',
+            'commented' => true,
+            'description' => 'Enables startup logs, including diagnostic checks',
+        ],
+        [
+            'name' => 'datadog.service',
+            'default' => 'my_service',
```

Comment:
I'm not sure if it's the agent or the backend, but _something_ will default to this service name if we manage to send something without a service:

```suggestion
            'default' => 'unnamed-php-service',
```

cc @brettlangdon , as this was mentioned somewhat recently in a discussion.